sycl/test-e2e/Config/env_vars.cpp

```diff
@@ -29,12 +28,10 @@ int main() {
         auto B = buf.get_access<access::mode::read_write>(cgh);
         cgh.single_task<class kernel_name1>([=]() { B[0] = 0; });
       });
-    } catch (sycl::runtime_error &e) {
+    } catch (sycl::exception &e) {
```

Comment:
I have a WIP patch that does this instead:
```
--- a/sycl/test-e2e/Config/env_vars.cpp
+++ b/sycl/test-e2e/Config/env_vars.cpp
@@ -33,8 +33,9 @@ int main() {
       // Exit immediately, otherwise the buffer destructor may actually try to
       // enqueue the command once again, and throw another exception.
       exit(0);
-    } catch (sycl::compile_program_error &e) {
-      exit(0);
+    } catch (sycl::exception &e) {
+      if (e.code() == errc::build)
+        exit(0);
     }
     assert(0 && "Expected exception was *not* thrown");
   } else {
```

I'm only changing one exception and not the other because I'm removing them one by one. I think you can either revert the change or check for both `errc::build` and `errc::runtime` error codes (and fail if not one of the two).